mobius-core/src/main/java/com/spotify/mobius/runners/ExecutorServiceWorkRunner.java

```diff
@@ -52,7 +52,7 @@ public void dispose() {
             "Disposing ExecutorServiceWorkRunner with {} outstanding tasks.", runnables.size());
       }
 
-      if (!service.awaitTermination(100, TimeUnit.MILLISECONDS)) {
+      if (!service.awaitTermination(Long.MAX_VALUE, TimeUnit.NANOSECONDS)) {
```

Comment:
I agree with @anawara, this is not safe. It would be good to understand why you want to make this change; it could be simply that the previous timeout of 100ms is too short in practice? A timeout is necessary in order to ensure that the dispose continues (and reports that there was a problem) rather than just stop silently, which is the behaviour this change leads to.

Reply:
This has been removed from the PR